src/content/docs/guides/Graphics/3-Graphical Application - Graphical-operation-management.mdx

```diff
@@ -0,0 +1,378 @@
+    newShape.color = "Red";   // Set default color
+    newShape.type = "Rectangle";  // Set default type to Rectangle
+    shapes.push_back(newShape);   // Add the new shape to the shapes container
+}
+```
+</TabItem>
+</Tabs>
+
+### **Shape Management and Rendering**
+Managing the rendering of shapes can be achieved by iterating over the shapes collection and calling the appropriate drawing function based on each shape's type. This enables the application to support multiple shape types.
+<Tabs syncKey="code-language">
+<TabItem label="C++">
+
+```cpp
+clear_screen(COLOR_WHITE);
+
+for (const auto& shape : shapes)
+{
+    fill_rectangle(COLOR_RED, shape.x, shape.y, shape.width, shape.height);
+}
+
+refresh_screen(60);
+```
+</TabItem>
+</Tabs>
+
+## Graphical operations 
+In this section, we’ll discuss how to handle the selection and movement of shapes. The code is designed to allow users to select shapes, move them around the canvas, and resize them. This approach makes the code more modular, easier to maintain, and extend.
+
+### **Detect Shape Selection**
+  First, we need to determine which shape the user has clicked on. This is achieved by checking if the mouse click position falls within the bounds of any existing shape. The `get_shape_at()` function is used to detect whether the mouse click is within a shape, allowing the program to identify which shape, if any, the user has selected.
+
+<Tabs syncKey="code-language">
+<TabItem label="C++">
+
+```cpp
+ShapeModel* get_shape_at(vector<ShapeModel>& shapes, double x, double y)
+{
+    for (auto& shape : shapes)
+    {
+        if (shape.type == "Rectangle" && x >= shape.x && x <= (shape.x + shape.width) && y >= shape.y && y <= (shape.y + shape.height))
+        {
+            return &shape;
+        }
+    }
+    return nullptr; // Return null if no shape is selected
+}
+```
+</TabItem>
+</Tabs>
+
+
+### **Dragging Shapes**
+  Once a shape is selected, the user can drag it by moving the mouse. This is done by updating the shape's x and y coordinates to follow the mouse's movement. The `is_moving` flag is used to distinguish between creating a new shape and moving an existing shape.
+
+<Tabs syncKey="code-language">
+<TabItem label="C++">
+
+```cpp
+if (is_moving && selected_shape != nullptr)
+{
+    selected_shape->x = mouse_x() - selected_shape->width / 2;
+    selected_shape->y = mouse_y() - selected_shape->height / 2;
+}
+```
+</TabItem>
+</Tabs>
+  
+### **Resizing Shapes**
+  In addition to dragging, users can resize shapes. When the user holds down the SPACE key and clicks on a shape, resizing mode is activated. The shape's width and height are updated based on the mouse position.
+
+<Tabs syncKey="code-language">
+<TabItem label="C++">
+
+```cpp
+if (is_resizing && selected_shape != nullptr)
+{
+    selected_shape->width = mouse_x() - selected_shape->x;
+    selected_shape->height = mouse_y() - selected_shape->y;
+}
+
+```
+</TabItem>
+</Tabs>
+
+### **Release pattern**
```

Comment:
Update this to something like "Final Shape Position" or "Finishing Shapes" etc. (To avoid confusion about "pattern" and focus on the outcome of releasing the mouse button rather than the action of this.)